runners/flink/src/test/java/org/apache/beam/runners/flink/website/PipelineOptionsTableGenerator.java

```diff
@@ -0,0 +1,179 @@
+    }
+  }
+
+  public static void main(String[] args) {
+    if (args.length != 1) {
+      throw new RuntimeException(
+          "Please specify the language (" + supportedLanguages + ") as the only argument.");
+    }
+    String arg = args[0].toLowerCase();
+    if (!supportedLanguages.contains(arg)) {
+      throw new RuntimeException("The language is not supported: " + arg);
+    }
+    boolean isPython = arg.equals("python");
+
+    printHeader();
+    List<Option> options = extractOptions(isPython);
+    printOptionsTable(options);
+  }
+
+  private static void printHeader() {
+    System.out.println(
+        "<!--\n"
+            + "Licensed under the Apache License, Version 2.0 (the \"License\");\n"
+            + "you may not use this file except in compliance with the License.\n"
+            + "You may obtain a copy of the License at\n"
+            + "\n"
+            + "http://www.apache.org/licenses/LICENSE-2.0\n"
+            + "\n"
+            + "Unless required by applicable law or agreed to in writing, software\n"
+            + "distributed under the License is distributed on an \"AS IS\" BASIS,\n"
+            + "WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.\n"
+            + "See the License for the specific language governing permissions and\n"
+            + "limitations under the License.\n"
+            + "-->");
+    System.out.println(
+        "<!--\n"
+            + "This is an auto-generated file.\n"
+            + "Use generatePipelineOptionsTableJava and generatePipelineOptionsTablePython respectively\n"
+            + "which should be called before running the tests.\n"
+            + "-->");
+  }
+
+  /**
+   * Returns the extracted list of options via reflections on FlinkPipelineOptions. Options are
+   * returned sorted in alphabetical order since Java does not guarantee any consistent order on the
+   * class methods.
+   */
+  private static List<Option> extractOptions(boolean isPython) {
+    List<Option> options = new ArrayList<>();
+    for (Method method : FlinkPipelineOptions.class.getDeclaredMethods()) {
+      String name;
+      String description;
+      String defaultValue = null;
+      name = method.getName();
+      if (name.matches("^(get|is).*")) {
+        name = name.replaceFirst("^(get|is)", "");
+
+        if (isPython) {
+          name = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_UNDERSCORE, name);
+        } else {
+          name = Character.toLowerCase(name.charAt(0)) + name.substring(1);
+        }
+
+        Description descriptionAnnotation = method.getAnnotation(Description.class);
+        if (descriptionAnnotation == null) {
+          throw new RuntimeException(
+              "All pipeline options should have a description. Please add one for " + name);
+        }
+        description = descriptionAnnotation.value();
+
+        Optional<String> defaultValueFromAnnotation = getDefaultValueFromAnnotation(method);
```

Comment:
I'm not sure we should use Optional here if we pass the null further to Options() anyway. Notice that in case when optional is empty the defaultValue is assigned to null anyway

Reply:
`Optional` is used for the return value. It is generally discouraged to use `Optional` on class fields. I agree that we could also return null here, it doesn't matter much though :)